src/commands.rs

```diff
@@ -398,15 +398,17 @@ impl<'a, V: SelectValue> KeyValue<'a, V> {
         }
     }
 
-    pub fn obj_len(&self, path: &str) -> Result<usize, Error> {
-        let first = self.get_first(path)?;
-        match first.get_type() {
-            SelectValueType::Object => Ok(first.len().unwrap()),
-            _ => Err(
-                err_msg_json_expected("object", self.get_type(path).unwrap().as_str())
-                    .as_str()
-                    .into(),
-            ),
+    pub fn obj_len(&self, path: &str) -> Result<ObjectLen, Error> {
+        match self.get_first(path) {
+            Ok(first) => match first.get_type() {
+                SelectValueType::Object => Ok(ObjectLen::Len(first.len().unwrap())),
+                _ => Err(
+                    err_msg_json_expected("object", self.get_type(path).unwrap().as_str())
+                        .as_str()
+                        .into(),
+                ),
+            },
+            _ => Ok(ObjectLen::NoneExisting),
```

Comment:
Doesn't it make more sense that this is an error?

Reply:
Yes, in new behavior it is an error (see `json_obj_len`). This is to maintain legacy behavior.

Reply:
So why add it as part of the valid return value and not as error?

Reply:
Just for backward compatibility with V 1.0 (only with legacy path)
`JSON.OBJKEYS doc1 .nowhere` and `JSON.OBJKEYS no_such_doc .` should return null (as before)